### DISCUSSION TITLE
Netbox 4.0 & Napalm plugin

### DISCUSSION THREAD

**Discussion by andy-butterworth** [netbox-community/netbox#16023] [Help Wanted!]
I've built a new Ubuntu 24.04 LTS server to replace a 20.04 server that currently runs NetBox 3.7.5.  The python 3.10+ requirements forced me to build a new server rather than upgrade the original.  Anyway its built and NetBox 4.0 is installed and working.  I copied the various settings from the old host to the new one by updating the configuration.py file (for the LDAP auth, Napalm plugin, SMTP etc), adding the ldap_config.py and local_requirements.txt files.

LDAP is working OK, as is SMTP, however the Napalm plugin isn't.  It appears as a plugin on the bottom left of the GUI and I can see the tabs for 'Status', 'LLDP Neighbors' and 'Config' when I've added a device with a management address etc.  However there is no data when I open any of the tabs.  I've done a bit of debugging on the device I've added and there are no SSH connection attempts.

I have tested from the CLI in the venv with the command 'napalm --user username --vendor ios 192.168.255.251 call get_facts' and this works OK.  I'm missing something, but not sure what.  The instructions for adding the Napalm plugin are fairly simple, so I'm not sure what I've missed.

**Comment by davidemiccone** [comment]
Hello Andy,
have you resolved?
Mee too I want to update to 24.04 (instead of 22.04).

**Reply by andy-butterworth** [reply]
Yes.  I built a new 24.04 Ubuntu host and installed NetBox and the Napalm plugin.  Initially the Napalm 0.2.0 plugin didn't work, however the 0.2.1 plugin works.  https://github.com/netbox-community/netbox-napalm-plugin

**Reply by candlerb** [reply]
> however the 0.2.1 plugin works

And has a [minimum Netbox version](https://github.com/netbox-community/netbox-napalm-plugin/blob/v0.2.1/netbox_napalm_plugin/__init__.py#L24) requirement of v4.0.2 - best to upgrade straight to v4.0.7 if you haven't already.

**Reply by andy-butterworth** [reply]
Sorry, yes.  There is a compatibility matrix on the netbox-napalm github repo
